src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -223,6 +225,13 @@ export class Kernel implements IKernel {
         if (!this.notebook) {
             return;
         }
+
+        // Set the notebook property on the matching editor
+        const editor = this.editorProvider.editors.find((item) => this.fs.arePathsSame(item.file, this.uri));
```

Comment:
I needed this property as I was looking at editor.notebook?.getMatchingInterpreter to get the candidate interpreter to use. However editor.notebook was never getting set for native notebooks. It was always undefined. 

As a side effect this code here wasn't actually doing anything before (as notebook was always undefined):
https://github.com/microsoft/vscode-jupyter/blob/cff6406f7d942b13fdc8dd16688ff9c376aa43da/src/client/datascience/notebook/notebookEditorProvider.ts#L133

So this is the side effect that I mentioned at the meeting that native notebooks will actually dispose their notebooks now when closed. Before they would only dispose on vscode close.